synapse/lib/types.py

```diff
@@ -613,6 +614,55 @@ def _normPyTuple(self, valu):
     def indx(self, norm):
         return s_common.buid(norm)
 
+class Hex(Type):
+    _opt_defs = (
+        ('width', 0),
+    )
+
+    def postTypeInit(self):
+        self._width = self.opts.get('width')
+        if self._width < 0:
+            # zero means no width check
+            raise s_exc.BadConfValu(name='width', valu=self._width,
+                                    mesg='Width must be > 0')
+        self._bsize = int(math.ceil(self._width / 2))
+        self._regex = regex.compile(r'^[0-9a-f]+$')
+        self.setNormFunc(str, self._normPyStr)
+        self.setNormFunc(int, self._normPyInt)
+        self.setNormFunc(bytes, self._normPyBytes)
+
+    def _normPyStr(self, valu):
+        valu = valu.strip().lower()
+        if valu.startswith('0x'):
+            valu = valu[2:]
+
+        if not valu:
+            raise s_exc.BadTypeValu(valu=valu,
+                                    mesg='no valu left?')
+
+        if not self._regex.match(valu):
+            raise s_exc.BadTypeValu(valu=valu,
+                                    mesg='regex fail')
+        if self._width and len(valu) != self._width:
```

Comment:
I would check if self._width is not None unless we change the above check so it can't be 0